exfor_parserpy/exfor_parser.py

```
@@ -248,14 +252,14 @@ def output_subentry(datadic, ofs=0, auxinfo=None):
 
 
 def parse_entry(lines, ofs=0, auxinfo=None, parse_opts=None):
-    datadic = {"subentries": []}
     if read_str_field(lines[ofs], 0) != "ENTRY":
         raise TypeError("not an ENTRY block")
     if auxinfo is None:
         auxinfo = {}
     auxinfo["entryid"] = read_str_field(lines[ofs], 1).strip()
     ofs += 1
     datadic = {}
+    datadic["__lastmodified"] = read_str_field(lines[ofs], 2, 1).strip()
```

Shouldn't this instruction be before `ofs += 1` to ensure it reads the N2 from the ENTRY line?

---

Good catch, it seemed in many files those two fields are identical, so it would have been tricky to find this bug later.

Regarding the formatting of this field, in one of the test files it does not follow the standard `yyyymmdd` format as the field contains [801103](https://github.com/IAEA-NDS/exfor-parserpy/blob/a790907dcad78f72238358b83e2d7f5427465c8e/tests/testdata/entry_21308.txt#L1). Any idea if this is supposed to be `yymmdd`, `ddmmyy` or something completely different. Would it be better to raise an error / warning in this case?

---

Thanks for rearranging the lines. Now we still need to deal with the `search_for_field` issue in `output_entry` and then I would suggest to merge and close this PR.

---

Maybe I missunderstood. What changes besides https://github.com/IAEA-NDS/exfor-parserpy/pull/7/commits/bbc94ee1c15e4f15281bce0db713b91b72491b64 are you asking for?

---

Time-wise, your commit crossed my comment, thanks for implementing it!